### PR TITLE
Kamila/lastfm_genre_api

### PR DESCRIPTION
**Pull request by kamilababayeva**
### **LastfmGenre API**

This PR is related to issue https://github.com/MaximeZmt/SDP_2022-Vibester/issues/70

This PR contains:

**ch.sdp.vibester.api: LastfmApi**: an api with a function querySongList(...) that will return list of songs within a given tag/genre
**ch.sdp.vibester.api: LastfmHelper**: a helper to combine songs from two Lastfm queries. This is done to have random songs each game.
**ch.sdp.vibester.api: LastfmUri**: a data class to build Uri to lastfm queries
**ch.sdp.vibester.model**: SongList: a class that represents an object retrieved from lastfm query
**ch.sdp.vibester**: GenreTemporary: an activity that is used to test the Genre API

In addition, the code is tested with a coverage of: [92]%

image

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/GenreTemporary.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package ch.sdp.vibester
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.widget.ArrayAdapter
+import android.widget.ListView
+import ch.sdp.vibester.api.LastfmHelper
+
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/GenreTemporary.kt:9]: Some javadoc to describe the class would be great. Maybe the method too.

Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/GenreTemporary.kt:9]: Then it's temporary so not mandatory

File: app/src/main/java/ch/sdp/vibester/model/SongList.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package ch.sdp.vibester.model
+
+import org.json.JSONArray
+import org.json.JSONObject
+import java.lang.IllegalArgumentException
+
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/model/SongList.kt:6]: Maybe adding a class doc ?

Comment (MaximeZmt) [reply app/src/main/java/ch/sdp/vibester/model/SongList.kt:6]: What does a song List represent. Also not a List<Song>. Doc may be important

**Review by MaximeZmt — COMMENTED** [review]
Really Nice code 👍 
There are some minor doc changes to do, but it should take seconds to solve.

**Review by MaximeZmt — APPROVED** [review]
Nice work 👍

File: app/src/main/java/ch/sdp/vibester/model/SongList.kt (hunk 2 above)
Comment (zwierski) [reply app/src/main/java/ch/sdp/vibester/model/SongList.kt:6]: The Javadoc looks good now

**Review by zwierski — APPROVED** [review]
Looks good to me. The code is well-documented.

**Pull request merged**